crates/fuel-core/src/query/block.rs

```diff
@@ -46,7 +46,7 @@ pub trait BlockQueryData: Send + Sync + SimpleBlockData {
 
 impl<D> BlockQueryData for D
 where
-    D: OnChainDatabase + DatabaseBlocks + ?Sized,
```

Comment:
why are we removing DatabaseBlocks from here?

Reply:
Two answers depending on the interpretation of the question:

_Why is it valid to remove DatabaseBlocks from here?_
Because the `OnChainDatabase` trait already requires `DatabaseBlocks` (see `crates/fuel-core/src/graphql_api/ports.rs`)

_Why are we removing DatabaseBlocks from here **in this PR**?_
Because of the "Boy Scout rule" - This was a small improvement that I preferred to clean up as I spotted it rather than create a standalone PR or issue for it.

Though looking at this again, I think `DatabaseBlocks` is the only required trait here so I think we can make this even cleaner by dropping the `OnChainDatabase` requirement here instead.

Reply:
Changed this so we're removing `OnChainDatabase` instead of `DatabaseBlocks`.

Reply:
nice, didn't see that ~ thanks!